group_badges.go

```diff
@@ -0,0 +1,206 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// GroupBadge represents a group badge.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/group_badges.html
+type GroupBadge struct {
+	ID               int    `json:"id"`
+	LinkURL          string `json:"link_url"`
+	ImageURL         string `json:"image_url"`
+	RenderedLinkURL  string `json:"rendered_link_url"`
+	RenderedImageURL string `json:"rendered_image_url"`
+	// Kind represents a project badge kind. Can be empty, when used PreviewProjectBadge().
+	Kind string `json:"kind"`
```

Comment:
I think it would be nice if we create a specific type for this that represents the kind. Something like we did here: https://github.com/xanzy/go-gitlab/blob/master/gitlab.go#L79-L90

And then maybe call it `BadgeKind` or something? That would make this field a little more strongly types. Through I now notice that we never get to set this field, but we only read it from the API. So that maybe makes this a bit less relevant. What do you think?

Reply:
Makes sense. I've introduced a type for badge kind.